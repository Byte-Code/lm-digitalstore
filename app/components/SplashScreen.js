import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Link } from 'react-router';
import { Map } from 'immutable';
import glamorous from 'glamorous';

import ScreenSaver from './ScreenSaver';

const Wrapper = glamorous.div({
  display: 'flex',
  height: '100%',
  width: '100%',
  flexWrap: 'wrap'
});

const BottomDiv = glamorous.div({
  width: '100%',
  alignSelf: 'flex-end',
  display: 'flex',
  flexDirection: 'column'
});

export default class Page extends Component {
  static propTypes = {
    world: ImmutablePropTypes.map.isRequired,
    requestFetchWorld: PropTypes.func.isRequired
  };

  static defaultProps = {
    forecast: Map()
  };

  componentDidMount() {
    this.props.requestFetchWorld();
  }

  render() {
    const { world } = this.props;

    const screenSaverVideo = world && world.get('screenSaverVideo', null);

    if (!screenSaverVideo) {
      return null;
    }

    return (
      <Link to="/world">
        <Wrapper>
          <BottomDiv>
            <ScreenSaver video={screenSaverVideo} />
          </BottomDiv>
        </Wrapper>
      </Link>
    );
  }
}
