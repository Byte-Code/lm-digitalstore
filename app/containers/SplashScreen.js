import { connect } from 'react-redux';

import SplashScreen from '../components/SplashScreen';
import { requestFetchWeather } from '../actions/weatherActions';
import { requestFetchWorld } from '../actions/worldActions'
import { getWeather , getWorld } from '../reducers/selectors';
import { startAnalyticsSession } from '../actions/analyticsActions';

const mapStateToProps = (state) => ({
  forecast: getWeather(state),
  world : getWorld(state)
});

const mapDispatchToProps = {
  requestFetchWeather,
  startAnalyticsSession,
  requestFetchWorld
};

export default connect(mapStateToProps, mapDispatchToProps)(SplashScreen);
