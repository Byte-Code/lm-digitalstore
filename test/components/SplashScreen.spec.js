import React from 'react';
import { shallow, mount } from 'enzyme';
import { Map, fromJS } from 'immutable';

import SplashScreen from '../../app/components/SplashScreen';

const world = fromJS({ screenSaverVideo: 'theVideoUrl' });
const requestFetchWorld = jest.fn();


describe('SplashScreen', () => {
  it('should return null when world is null is empty', () => {
    const result = shallow(
      <SplashScreen
        requestFetchWorld={requestFetchWorld}
        world={Map()}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should otherwise render properly', () => {
    const result = shallow(
      <SplashScreen
        world={world}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should call requestFetchWorld after mounting', () => {
    mount(
      <SplashScreen
        world={world}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(requestFetchWorld).toHaveBeenCalled();
  });
});
