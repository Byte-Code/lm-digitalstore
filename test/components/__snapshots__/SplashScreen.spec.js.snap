// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen should otherwise render properly 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/world"
>
  <glamorous(div)>
    <glamorous(div)>
      <ScreenSaver
        video="theVideoUrl"
      />
    </glamorous(div)>
  </glamorous(div)>
</Link>
`;

exports[`SplashScreen should return null when world is null is empty 1`] = `null`;
